package.json

```diff
@@ -107,6 +107,7 @@
     "doctrine": "^2.1.0",
     "eslint-import-resolver-node": "^0.3.7",
     "eslint-module-utils": "^2.7.4",
+    "fast-deep-equal": "^3.1.3",
```

Comment:
i'm very hesitant to add this dependency. `deep-equal` is the minimal amount of code required for correctness - i'd prefer to use that package, and if *it* could use some perf optimizations then that'd be great.

Reply:
@ljharb 

> `deep-equal` is the minimal amount of code required for correctness

I think this is not true, as `deep-equal` [does handle way more cases than needed](https://unpkg.com/deep-equal@2.2.0/index.js), e.g. comparison of ArrayBuffers and whatnot. Furthermore it has  has 43 dependencies: https://npm.anvaka.com/#/view/2d/deep-equal and on my machine is even slower than `assert.deepStrictEqual`:

```
fast-deep-equal x 216,001 ops/sec ±0.37% (92 runs sampled)
fast-equals x 227,635 ops/sec ±0.44% (93 runs sampled)
nano-equal x 144,366 ops/sec ±0.46% (91 runs sampled)
shallow-equal-fuzzy x 114,739 ops/sec ±0.38% (91 runs sampled)
underscore.isEqual x 74,419 ops/sec ±0.46% (92 runs sampled)
lodash.isEqual x 34,833 ops/sec ±0.39% (91 runs sampled)
deep-equal x 96.92 ops/sec ±1.82% (55 runs sampled)
deep-eql x 21,870 ops/sec ±1.87% (89 runs sampled)
ramda.equals x 11,929 ops/sec ±3.34% (88 runs sampled)
util.isDeepStrictEqual x 46,825 ops/sec ±1.26% (88 runs sampled)
assert.deepStrictEqual x 515 ops/sec ±0.75% (88 runs sampled)
```

If we have a look at what `hashObject` / `hashify` is actually capable of hashing / differentiating (Array, 0bject, stuff that is stringifyable with `JSON.stringify`), the [simplicity of `fast-deep-equal`](http://unpkg.com/browse/fast-deep-equal@3.1.3/index.js) actually aligns quite well with the `hashObject` implementation.

- - -

That being said, what do you think of moving the memoization and comparison to a new util, e.g. `hashObjectMemoized` which would be used like this:

```
    cacheKey: String(parserPath) + hashObjectMemoized(parserOptions, 'parserOptions') + hashObjectMemoized(settings, 'settings') + String(path),
```

That way if we ever change the implementation of `hashify`/`hashObject`, we could adjust the `isEqual` function for `hashObjectMemoized` to align with the implementation of the hashing algorithm. For example if support for hashing Sets or Maps is added in the future.

Reply:
Another note, I would probably go with:

a) Custom equality implementation that matches `hashObject` needs
b) [`fast-equals`](https://github.com/planttheidea/fast-equals) over `deep-equal` or `fast-deep-equal`, because: 0 dependencies, maintained, future proof and the fastest of the bunch, as seen above and via their [own benchmarks](https://github.com/planttheidea/fast-equals#benchmarks)